src/call_builder.ts

```diff
@@ -213,21 +247,20 @@ export class CallBuilder {
    * @param {object} json Response object
    * @returns {object} Extended response object
    */
-  _toCollectionPage(json) {
+  private _toCollectionPage(json: any): any {
     for (let i = 0; i < json._embedded.records.length; i += 1) {
-      // eslint-disable-next-line no-param-reassign
       json._embedded.records[i] = this._parseRecord(json._embedded.records[i]);
     }
     return {
       records: json._embedded.records,
-      next: () =>
-        this._sendNormalRequest(URI(json._links.next.href)).then((r) =>
-          this._toCollectionPage(r)
-        ),
-      prev: () =>
-        this._sendNormalRequest(URI(json._links.prev.href)).then((r) =>
-          this._toCollectionPage(r)
-        )
+      next: async () => {
+        const r = await this._sendNormalRequest(URI(json._links.next.href));
+        return this._toCollectionPage(r);
+      },
+      prev: async () => {
+        const r = await this._sendNormalRequest(URI(json._links.prev.href));
+        return this._toCollectionPage(r);
+      }
```

Comment:
In this case, I think the change is less concise without any change in clarity, so I think it was better as it was before.